rust_src/src/buffers.rs

```diff
@@ -590,7 +582,7 @@ impl From<LispObject> for LispBufferOrName {
         if v.is_string() {
             LispBufferOrName::Name(v)
         } else {
-            v.as_buffer_or_error();
+            LispBufferRef::from(v);
```

Comment:
This is another instance of spurious casting. `check_buffer` would be cheaper.

Reply:
Maybe it would be better to make that an `else if` and make the final `else` a `wrong_type!` or something like that.